### PR TITLE
Tidy driver/database-supports? defmethods and remove usage of driver/supports?

### PR DESCRIPTION
**Pull request by calherries**
This is a refactor to make the use of `metabase.driver/database-supports?` more consistent and remove uses of `metabase.driver/supports?`, which should be deprecated. I did it as a mindless tidy up while I waiting for CI.

The changes fall into three categories, enumerated in the comments.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29516)

### CODE REVIEW

File: modules/drivers/athena/src/metabase/driver/athena.clj [hunk 1]
```diff
@@ -37,17 +37,11 @@
...
-                                                                         ; for now.
-
-(defmethod driver/database-supports? [:athena :test/jvm-timezone-setting]
-  [_driver _feature _database]
-  false)
+(doseq [[feature supported?] {:foreign-keys              true
+                              :datetime-diff             true
+                              :nested-fields             false #_true ; huh? Not sure why this was `true`. Disabled for now.
+                              :test/jvm-timezone-setting false}]
+  (defmethod driver/database-supports? [:athena feature] [_driver _feature _db] supported?))
```
Comment (calherries) [inline modules/drivers/athena/src/metabase/driver/athena.clj:44]: 1. Consolidation of several `driver/supports?` or `driver/database-supports?` defmethods into one `doseq`, where possible.

File: src/metabase/driver/postgres.clj [hunk 2]
```diff
@@ -46,44 +46,34 @@
...
-
 (doseq [feature [:actions :actions/custom]]
   (defmethod driver/database-supports? [:postgres feature]
     [driver _feat _db]
     ;; only supported for Postgres for right now. Not supported for child drivers like Redshift or whatever.
     (= driver :postgres)))
 
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                             metabase.driver impls                                              |
+;;; +----------------------------------------------------------------------------------------------------------------+
```
Comment (calherries) [inline src/metabase/driver/postgres.clj:73]: In most namespaces, the `driver/database-supports?` block is above the metabase.driver impls block

**Review by camsaul — APPROVED** [review]
Awesome, thanks for doing this.

Can you add a note to the driver changelog in `docs/` about this method actually being deprecated for real now, and our timeline for removing it completely?

**Review by qnkhuat — APPROVED** [review]

**Comment by calherries** [comment]
@camsaul I've added a note in the changelog here. https://github.com/metabase/metabase/commit/70bd78cd3ecf4c7bb17caedc2ccff7fa5fafe3de

**Pull request merged**